lib/sandbox.macos.ml

```diff
@@ -115,14 +123,16 @@ let run ~cancelled ?stdin:stdin ~log (t : t) config result_tmp =
     post_build ~result_dir ~home_dir t >>= fun () ->
     Lwt.return r
   in
+  let () = Lwt.on_failure cancelled (fun _ -> let _ = unmount_fuse t in () )
```

Comment:
The returned promise most likely needs to be passed to `Lwt.async` to be executed by Lwt, just like the function passed to `Lwt.on_termination` below.